src/Components/test/testassets/TestServer/Pages/ConnectionQualityHost.cshtml

```diff
@@ -0,0 +1,40 @@
+@page
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Refresh tests</title>
+    <base href="~/" />
+</head>
+<body>
+    @(await Html.RenderComponentAsync<TestServer.Refresh>())
+
+    <button id="refresh-page" onclick="location.reload()">Refresh</button>
+
+    @*
+        So that E2E tests can make assertions about both the prerendered and
+        interactive states, we only load the .js file when told to.
+    *@
+    <hr />
+
+    <button id="load-boot-script" onclick="start()">Load boot script</button>
+
+    <script src="_framework/blazor.server.js" autostart="false"></script>
+    <script>
+        // Used by InteropOnInitializationComponent
```

Comment:
Copy-paste artifact?

Also it would be great not to be duplicating the boilerplate we see here from L22-L38. Could this be factored out into some static `.js` file shared among the E2E test cases that use it?

Reply:
Same with file below

Reply:
That's fair.